tests/test_mongodb_id_pk_variations.py

```diff
@@ -0,0 +1,228 @@
+                }
+
+    def name(self):
+        return "tap_tester_mongodb_id_pk_variations"
+
+    def tap_name(self):
+        return "tap-mongodb"
+
+    def get_type(self):
+        return "platform.mongodb"
+
+    def get_credentials(self):
+        return {'password': os.getenv('TAP_MONGODB_PASSWORD')}
+
+    def get_properties(self):
+        return {'host': os.getenv('TAP_MONGODB_HOST'),
+                'port': os.getenv('TAP_MONGODB_PORT'),
+                'user': os.getenv('TAP_MONGODB_USER'),
+                'database': os.getenv('TAP_MONGODB_DBNAME')
+                }
+
+    def test_run(self):
+
+        conn_id = connections.ensure_connection(self)
+
+        #  -------------------------------
+        # -----------  Discovery ----------
+        #  -------------------------------
+
+        # run in discovery mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        # verify the tap discovered the right streams
+        found_catalogs = menagerie.get_catalogs(conn_id)
+
+        # assert we find the correct streams
+        self.assertEqual(self.expected_check_streams(),
+                         {c['tap_stream_id'] for c in found_catalogs})
+
+        #  -----------------------------------
+        # -----------Initial Full Table Sync ---------
+        #  -----------------------------------
+        # Select simple_coll_1 and simple_coll_2 streams and add replication method metadata
+        for stream_catalog in found_catalogs:
+            annotated_schema = menagerie.get_annotated_schema(conn_id, stream_catalog['stream_id'])
+            additional_md = [{"breadcrumb": [], "metadata": {'replication-method': 'FULL_TABLE'}}]
+            selected_metadata = connections.select_catalog_and_fields_via_metadata(conn_id,
+                                                                                   stream_catalog,
+                                                                                   annotated_schema,
+                                                                                   additional_md)
+
+        runner.run_sync_mode(self, conn_id)
+
+        # streams that we synced are the ones that we expect to see
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        records_by_stream = runner.get_records_from_target_output()
+
+        # verify if we are capturing all the data for all the streams
+        self.assertEqual(record_count_by_stream, self.expected_record_count())
```

Comment:
I think there should be more to this test. 

- Validate the values and datatypes of the primary key rows in the target.
- Validate that the _id field is properly marked in metadata as the `table-key-property`.
- Do this test for multiple replication methods, depending on the tap implementation these datatypes may be decoded differently depending on the method.

Reply:
done